source/engine/graphics/api/vulkan/ShaderBindingManager.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include "graphics/GraphicsConstants.h"
+#include "utility/Signal.h"
+
+#include "vulkan/vk_mem_alloc.hpp"
+
+#include <span>
+#include <unordered_map>
+
+namespace nc::graphics
+{
+using SetIndex = uint32_t;
+using BindingSlot = uint32_t;
+
+struct DescriptorSetLayout
```

Comment:
We should hash out a data layout for these that's more in line with the access pattern. `bindings` and `bindingFlags` are accessed at the same time by the same key, but stored in different maps. Same with `writesPerSet` and `isDirty` from `DescriptorSets`. There's some hoop jumping to get contiguous views over some of the data.